client/src/annoMatrix/query.ts

```diff
@@ -1,22 +1,51 @@
 import sha1 from "sha1";
 import { _dubEncURIComp } from "./fetchHelpers";
+import { Field } from "../common/types/entities";
 
 /**
  * Query utilities, mostly for debugging support and validation.
  */
 
+export type ComplexQuery = SummarizeQuery | WhereQuery;
+
+export type Query = string | ComplexQuery | (string | ComplexQuery)[];
```

Comment:
I typed this based on the comments in AnnoMatrix L245 but am not sure about the `(string | ComplexQuery)[] ` union type. Feedback appreciated!

Reply:
LGTM!

Reply:
It is not quite right.
* the routines in this file accept a singleton query (string | ComplexQuery)
* _fetch and fetch in `annoMatrix.ts` accept either a singleton query or a list of those.

I recommend that this file defile Query as `string | ComplexQuery`, and that the parameter typing for `annoMatrix.ts::fetch()` be changed to accept `Query | Query[]`.  See the second line of _fetch(), where it does:
>         const queries = Array.isArray(q) ? q : [q];

Reply:
Updated:

* [query.ts::Query](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/query.ts#L11)
* [annoMatrix.ts::fetch](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/annoMatrix.ts#L235)
* [annoMatrix.ts::_fetch](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/annoMatrix.ts#L489)